ubccr/xdmod
# Replace `Config` Usage w/ `XdmodConfiguration`

**I'm going to apologize up front for the size of this PR but it turns out we use configuration files fairly extensively. On the plus side the vast majority of changes you'll be seeing are the same, just in different places.**

## Description
This PR removes all usage `Xdmod\Config` that is not in a file that itself will be removed soon ( `aRole` I'm looking at you! ) and replaces it w/ the newly consolidated `XdmodConfiguration`. `XdmodConfiguration` is derived from the same base class, `Configuration\Configuration` as the code utilized by ETLv2. After this PR has been merged all that should be left is to remove `Xdmod\Config` and the its associated tests.

In general the changes found in this PR fit one of a few use cases:

**Standard Usage / Replacement**

*Old Code:*
```php
<?php

// instance of \Log or CCR\Log
$logger;

$config = Config::factory();

$specificConfig = $config['specific'];

// do stuff w/ $specificConfig and use $logger to log
```

*New Code:*
```php
<?php

$configFile = new \Configuration\XdmodConfiguration(
    'specific.json',
    CONFIG_DIR,
    $logger,
    array(
        'local_config_dir' => implode(DIRECTORY_SEPERATOR, array(CONFIG_DIR, 'specific.d))
    )
);
$configFile->initialize();

$specificConfig = json_decode($configFile->toJson(), true);

// do the same stuff w/ $specificConfig as in the old code.
```

**`getFilePath`|`getConfigDirPath` usage / replacement**

*Old Code:*
```php
<?php

$config = Config::factory();

$configFilePath = $config->getFilePath('config-file.json');

// Do something with $configFilePath
```

*New Code:*
```php
<?php

$configFilePath = implode(
    DIRECTORY_SEPARATOR,
    array(
      CONFIG_DIR,
      'config-file.json'
    )
);

// Do same things w/ $configFilePath
```

## Motivation and Context
Having two sets of code for working with Configuration files is not needed. This allows any further enhancement to the Configuration processing code to be utilized across the entire code base. 

## Tests performed
For Docker - OpenXDMoD Install / Upgrade:
- Unit Tests
- Component Tests
- Integration Tests
- UI Tests
- REG_TEST_ALL=1 Regression Tests

Manual Testing:
- Log into the Internal Dashboard
- Ensure that the Summary Page displays correctly
- Ensure that the Log Tab displays correctly


## Types of changes

- [X] Code Refactor / Consolidation

## Checklist:
